app/helpers/api-util.js

```diff
@@ -1,10 +1,14 @@
 // TOURS
 import clientPromise from "@/components/mongo/client";
 
+async function getDbConnection() {
```

Comment:
this is the correct idea, however, we are still creating a db object every time this function is called

we should instantiate the db object once and reuse it everywhere

for that, we can create a separate file where we create the db object and export it for use in other modules